test/e2e/handler/lldp_with_nmpolicy_test.go

```diff
@@ -0,0 +1,78 @@
+/*
+Copyright The Kubernetes NMState Authors.
+
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package handler
+
+import (
+	"fmt"
+	"time"
+
+	nmstate "github.com/nmstate/kubernetes-nmstate/api/shared"
+	"github.com/nmstate/kubernetes-nmstate/test/e2e/policy"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("LLDP configuration with nmpolicy", func() {
+	lldpEnabledPolicyName := "lldp-enabled"
+	lldpDisabledPolicyName := "lldp-disabled"
+
+	configureLldpOnEthernetsCapture := func(enabled string) map[string]string {
+		return map[string]string{
+			"ethernets":      `interfaces.type=="ethernet"`,
+			"ethernets-up":   `capture.ethernets | interfaces.state=="up"`,
+			"ethernets-lldp": fmt.Sprintf(`capture.ethernets-up | interfaces.lldp.enabled:=%s`, enabled),
+		}
+	}
+
+	interfacesWithLldpEnabledState := nmstate.NewState(`interfaces: "{{ capture.ethernets-lldp.interfaces }}"`)
+
+	BeforeEach(func() {
+		By("Enabling LLDP on up ethernet interfaces")
+		setDesiredStateWithPolicyAndCapture(lldpEnabledPolicyName, interfacesWithLldpEnabledState, configureLldpOnEthernetsCapture("true"))
+
+		DeferCleanup(func() {
```

Comment:
We can put all the AfterEach on the DeferCleanup right ? also look slike we are repeting the DeferCleanup at the AfterEach.

Reply:
Added an example yaml, moved AfterEach into DeferCleanup